### ISSUE TITLE
[Feat]: Dynamic table structure

### ISSUE THREAD

**Issue by S0L1D3R** [lowcoder-org/lowcoder#1254]
**Feature**
Table component needs separate settings for when columns/data structure changes.

At minimum I'd like to be able to dynamically set:
- column label (title)
- column type
  - if not possible, it would be nice if pre-set type would remain in place when data structure changes. Currently, lowcoder regenerates table from 0 each time.
- column order
  -  currently this can be achieved by order of column visibility array but that is not easy to coordinate between multiple components

Ideally it would be nice to dynamically set other column settings available in the column settings panel or being able to split frontend filtering with backend.

**Use case**
I am trying to build personal project, based on ERP software I was using at my last company.
Basically, users could select columns they would like to see in the table and software would modify request to the backed so that only selected columns are returned to the frontend. (Currently I can only hide columns from the view, but data will be still requested and fetched)
Users could also select their own column order (left to right).
